src/bin/generate_wiki/main.rs

```diff
@@ -0,0 +1,18 @@
+use docs_builder::DocsBuilder;
+use std::{fs, path::PathBuf};
+use types::Component;
+
+mod docs_builder;
+mod types;
+
+fn main() {
+    let docs_path = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("docs");
+    if !docs_path.exists() {
+        fs::create_dir(&docs_path).unwrap();
+    }
+
+    let pages = DocsBuilder::new().add_definition::<Component>(true).build();
```

Comment:
given that you are creating a separate page for each component wouldn't be easier just to generate docs for each component explicitly?
e.g.
```
let schema = ...

let view = Docs::generate_component_docs(schema, "View", "view") // I assume you need that 3 argument to detect vairant?
let shader = DocsBuild::generate_compontnt_docs(schema, "Shader", "shader")

// write to file
```

This way you don't need to modify the schema

Also you don't seem to need the Builder abstraction, I assume this is leftover after some differnt approach?